src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -1161,6 +1162,17 @@ public bool TrueForAll(Predicate<T> match)
             return true;
         }
 
+        public ImmutableArray<T> ToImmutable()
```

Comment:
@sharwell has specific instructions for making changes to SegmentedList`1. I'm assuming that this change needs to be in a separate commit.

Reply:
Also, if we're going make this specific method, as opposed to just calling `ToImmutable` from the BCL or `SelectAsArray` with an identity selector, let's take advantage of array segments and copying entire chunks at a time, rather than just a `foreach` loop.

Reply:
@sharwell -- Should I move the SegmentedList.ToImmutable method to not be directly on SegmentedList?

Reply:
I have two problems with this:

1. This method needs to be named `ToImmutableArray()` to avoid confusion with `ImmutableSegmentedList<T>`
2. This method should be an extension method instead of a member method on the type